plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/OpenShiftServerExtendedProperties.java

```diff
@@ -48,6 +49,14 @@ public String getWelcomePageUrl() {
 		IProject project = service.getProject();
 		if (project != null) {
 			List<IRoute> routes = project.getResources(ResourceKind.ROUTE);
+			String url = OpenShiftServerUtils.getRouteURL(server);
+			if(!StringUtils.isEmpty(url)) {
+				for (IRoute route: routes) {
+					if(url.equals(route.getURL())) {
+						return welcomePageUrl = url;
+					}
+				}
+			}
```

Comment:
btw. we're loading routes here from the server, is this code invoked in a background job by the WTP framework? I see a  Display.syncExec hidden in the RouteChooser which is kinda unexpected, but I guess that's bcs it gets invoked at various places within jobs.